ui/guest/src/react/MoveModeZoneMenu.tsx

```diff
@@ -15,29 +15,76 @@
  */
 
 import * as React from 'react';
+import { useMemo } from 'react';
 import DragIndicatorRounded from '@mui/icons-material/DragIndicatorRounded';
 import ArrowDownwardRoundedIcon from '@mui/icons-material/ArrowDownwardRounded';
 import ArrowUpwardRoundedIcon from '@mui/icons-material/ArrowUpwardRounded';
 import DeleteOutlineRoundedIcon from '@mui/icons-material/DeleteOutlineRounded';
 import UltraStyledIconButton from './UltraStyledIconButton';
 import HighlightOffRoundedIcon from '@mui/icons-material/HighlightOffRounded';
 import { Tooltip } from '@mui/material';
-import { useDispatch } from './GuestContext';
-import * as elementRegistry from '../classes/ElementRegistry';
 import DragGhostElement from './DragGhostElement';
+import * as contentController from '../classes/ContentController';
+import { clearAndListen$ } from '../store/subjects';
+import { startListening } from '../store/actions';
+import { compileDropZone, fromElement, getParentElementFromICEProps } from '../classes/ElementRegistry';
+import { ElementRecord } from '../models/InContextEditing';
 
 export interface MoveModeZoneMenuProps {
   [key: string]: any;
+  record: ElementRecord;
 }
 
 export function MoveModeZoneMenu(props: MoveModeZoneMenuProps) {
   const { record, dispatch } = props;
-  // const dispatch = useDispatch();
+  const {
+    modelId,
+    fieldId: [fieldId],
+    index
+  } = record;
+
+  const elementIndex = useMemo(() => {
+    if (typeof index === 'string') {
+      return parseInt(index.substr(index.lastIndexOf('.') + 1), 10);
+    }
+    return index;
+  }, [index]);
+
+  const dropzoneRecord = useMemo(
+    () => fromElement(getParentElementFromICEProps(modelId, fieldId, index)[0] as Element),
+    [fieldId, index, modelId]
+  );
+
+  const dropzoneChildrenLength = useMemo(
+    () => compileDropZone(dropzoneRecord.iceIds[0]).children.length,
```

Comment:
If all you're getting from this is the number of children, it'd be best to not use the DOM and only use the ice registry to get this information.

Reply:
Try
```suggestion
    () => extractCollection(getCachedModel(modelId), fieldId, index).length,
```